crates/storage/src/blueprint/merklized.rs

```diff
@@ -305,260 +356,469 @@ where
     }
 }
 
-/// The macro that generates basic storage tests for the table with the merklelized structure.
-/// It uses the [`InMemoryStorage`](crate::structured_storage::test::InMemoryStorage).
 #[cfg(feature = "test-helpers")]
-#[macro_export]
-macro_rules! basic_merklelized_storage_tests {
-    ($table:ident, $key:expr, $value_insert:expr, $value_return:expr, $random_key:expr) => {
-        $crate::paste::item! {
-        #[cfg(test)]
-        #[allow(unused_imports)]
-        mod [< $table:snake _basic_tests >] {
-            use super::*;
-            use $crate::{
-                structured_storage::test::InMemoryStorage,
-                transactional::WriteTransaction,
-                StorageAsMut,
-            };
-            use $crate::StorageInspect;
-            use $crate::StorageMutate;
-            use $crate::rand;
-            use $crate::tables::merkle::DenseMetadataKey;
-            use rand::SeedableRng;
-
-            #[allow(dead_code)]
-            fn random<T, R>(rng: &mut R) -> T
-            where
-                rand::distributions::Standard: rand::distributions::Distribution<T>,
-                R: rand::Rng,
-            {
-                use rand::Rng;
-                rng.gen()
-            }
+/// A trait that provides basic tests for the merklized storage.
+/// It is used to test the merklized storage with different key and value codecs.
+#[allow(warnings)]
+pub mod basic_tests {
```

Comment:
```suggestion
pub mod basic_tests_bmt {
```

Reply:
Done in https://github.com/FuelLabs/fuel-core/pull/2914/commits/2769a274cef36534cf628d7da82c476b174ac578